3-UsingAPIs/Week1/homework/ex1-johnWho.js

```diff
@@ -7,20 +7,20 @@ Rewrite this function, but replace the callback syntax with the Promise syntax:
   didn't pass in a first name!"
 ------------------------------------------------------------------------------*/
 // TODO see above
-const getAnonName = (firstName, callback) => {
-  setTimeout(() => {
-    if (!firstName) {
-      callback(new Error("You didn't pass in a first name!"));
-      return;
-    }
+const getAnonName = (firstName) => {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      if (!firstName) {
+        reject(new Error("You didn't pass in a first name!"));
+      }
+      const fullName = `${firstName} Doe`;
 
-    const fullName = `${firstName} Doe`;
-
-    callback(fullName);
-  }, 1000);
+      resolve(fullName);
+    }, 1000);
+  });
 };
 
-getAnonName('John', console.log);
+getAnonName('John').then((response) => console.log(response));
```

Comment:
This could be simplified to:

```suggestion
getAnonName('John').then(console.log);
```

Reply:
Yes, This is indeed shorter and new for me 😅 . Is it here possible to do this, because we have only one parameter and want only to `return` console.log(that parameter)?
Thanks a lot. 👍

Reply:
The logic behind this simplification is as follows:

1. The `.then()` method takes one parameter, which must be a function.
2. It calls this function when the promise is resolved and passes the resolved value to that function as a parameter.
3. If you pass the  `console.log` method as a parameter to `.then()` this function will be called and gets as a parameter the resolved value.
4. `console.log` will output the resolved value to the console as we intended.

Reply:
Got it! many thanks a lot @remarcmij